airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/KafkaFormatFactory.java

```diff
@@ -0,0 +1,25 @@
+package io.airbyte.integrations.source.kafka;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.integrations.source.kafka.format.AvroFormat;
+import io.airbyte.integrations.source.kafka.format.JsonFormat;
+import io.airbyte.integrations.source.kafka.format.KafkaFormat;
+
+public class KafkaFormatFactory {
+
+    public static KafkaFormat getFormat(final JsonNode config){
+
+        MessageFormat messageFormat = config.has("MessageFormat")? MessageFormat.valueOf(config.get("MessageFormat").get("deserialization_type").asText().toUpperCase())
+                :MessageFormat.JSON;
+
+        switch (messageFormat) {
```

Comment:
We can later extend this out to include PROTOBUF support using confluent libraries - Solves #15059 and #15059